src/main/java/phase1/SentimentMapper.java

```diff
@@ -1,103 +1,104 @@
 package phase1;
 
-import static util.Constants.AUTHOR_INDEX;
-import static util.Constants.COMMENT_INDEX;
-import static util.Constants.COMMIT_DATE_INDEX;
-import static util.Constants.COMMIT_DATE_PATTERN;
-import static util.Constants.LANGUAGE_INDEX;
-
-import com.opencsv.CSVReader;
-import com.opencsv.exceptions.CsvMalformedLineException;
-import com.opencsv.exceptions.CsvValidationException;
 import edu.stanford.nlp.ling.CoreAnnotations;
 import edu.stanford.nlp.neural.rnn.RNNCoreAnnotations;
 import edu.stanford.nlp.pipeline.Annotation;
 import edu.stanford.nlp.pipeline.StanfordCoreNLP;
 import edu.stanford.nlp.sentiment.SentimentCoreAnnotations.SentimentAnnotatedTree;
 import edu.stanford.nlp.trees.Tree;
 import edu.stanford.nlp.util.CoreMap;
-import java.io.IOException;
-import java.io.StringReader;
-import java.time.LocalDate;
-import java.time.format.DateTimeFormatter;
-import java.time.temporal.WeekFields;
-import java.util.Locale;
-import java.util.Properties;
 import model.DateInfo;
 import model.PatternInfo;
 import model.PatternType;
 import model.SentimentInfo;
 import model.SentimentType;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
 import org.apache.hadoop.io.LongWritable;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapreduce.Mapper;
 
+import java.io.IOException;
+import java.io.StringReader;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.WeekFields;
+import java.util.Locale;
+import java.util.Properties;
+
 public class SentimentMapper extends Mapper<LongWritable, Text, PatternInfo, SentimentInfo> {
   private StanfordCoreNLP pipeline;
   private DateTimeFormatter formatter;
+
+  private CSVFormat csvFormat;
+
   @Override
   protected void setup(Mapper<LongWritable, Text, PatternInfo, SentimentInfo>.Context context) {
-    formatter = DateTimeFormatter.ofPattern(COMMIT_DATE_PATTERN);
+    formatter = DateTimeFormatter.ofPattern(util.Constants.COMMIT_DATE_PATTERN);
     Properties props = new Properties();
     props.setProperty("annotators", "tokenize, parse, sentiment");
     pipeline = new StanfordCoreNLP(props);
+    csvFormat=CSVFormat.EXCEL.builder()
+            .setRecordSeparator(',')
+            .setSkipHeaderRecord(true)
+            .setIgnoreSurroundingSpaces(true).build();
   }
```

Comment:
I have tested the 10000 record file with this implementation, and it worked fine, but as we know its not because of this code, rather its because of the formatted data. However, what I will do is revert it to CSVReader test the 10000 records again and revert this change if its is not needed